docs/faucets/dd.md

```diff
@@ -0,0 +1,363 @@
+
+	function mint(address _recipient) external override {
+		FEE_ASSET.mint(_recipient, mintAmount);
+	}
+
+	function setMintAmount(uint256 _amount) external override onlyOwner {
+    mintAmount = _amount;
+	}
+}
+```
+
+We will manually deploy this contract. The owner will be the same as the current owner of the FeeAsset.
+
+The handler's deployed address will need to be made known in discord and other public channels/docs.
+
+#### Choosing the mint amount
+
+As mentioned above, we only need to prevent infinite minting (i.e. over 2^256).
+
+We can calculate the maximum mint amount based on the expected L1 block production rate, the expected L1 gas cost of a mint, and the amount of time we expect the handler to be live.
+
+Assume the cost to mint is 5K gas.
+
+Then if a block were completely full of `mint` transactions, the maximum number of mints would be 38M / 5K = 7600.
+
+This means that we could be maximally minting 7600 / 12 = 633 times per second.
+
+Suppose that we expect the handler to be live for a maximum of 5 years.
+
+Then we have a maximum of 5 \* 365 \* 24 \* 60 \* 60 \* 633 ~= 1e11 mints.
+
+Thus, the maximum mint amount is (2^256 - 1) / 1e11 ~= 1e66.
+
+But we'll err on the side of caution and use 1e42.
+
+#### Alternative Design
+
+The primary alternative would be to make a frontend UI that connects to a backend that has the account that is the owner of the token, then maintain a map on the backend to rate-limit different IP addresses, or have users sign up, etc.
+
+The benefit is that if we went with the signup route, we could advertise to those users.
+
+The downside is that that is significantly more infrastructure, and it doesn't necessarily improve the safety/control of the system.
+
+### Staking Asset
+
+The staking asset will be deployed with the same owner as the fee asset. In addition, a `StakingAssetHandler` will be deployed, which will have the minter role on the staking asset.
+
+### Staking Asset Handler
+
+```solidity
+interface IStakingAssetHandler {
+	function addValidator(address _attester, address _proposer) external;
+	function setRollup(address _rollup) external;
+	function setDepositAmount(uint256 _amount) external;
+	function setMinMintInterval(uint256 _interval) external;
+}
+
+contract StakingAssetHandler is IStakingAssetHandler, Ownable {
+	IMintableERC20 public immutable STAKING_ASSET;
+
+	uint256 public depositAmount;
+	uint256 public lastMintTimestamp;
+	uint256 public minMintInterval;
+	IStakingCore public rollup;
+
+	constructor(address _owner, address _stakingAsset, address _rollup, uint256 _depositAmount, uint256 _minMintInterval) Ownable(_owner) {
+		STAKING_ASSET = IMintableERC20(_stakingAsset);
+		depositAmount = _depositAmount;
+		rollup = IStakingCore(_rollup);
+		minMintInterval = _minMintInterval;
+	}
+
+	function addValidator(address _attester, address _proposer) external override onlyOwner {
+		require(block.timestamp - lastMintTimestamp >= minMintInterval, "Min mint interval not met");
```

Comment:
This one makes sense to me if you are trying to drip people in one at a time, but not if you just want to limit growth daily. I'm not fully getting which is your target here?

If daily, you would mint if out of funds, and "reset" that clock, so you don't need to mint every time. 

Also, this will fail the deposit because it have not approved the rollup to pull funds.